### PR TITLE
Adding admin operations, segregating account dto and making account c…

### PR DESCRIPTION
**Pull request by evakule**
…reation limit

### CODE REVIEW

File: src/test/java/com/somn/service/AccountEntityServiceImplTest.java [hunk 1]
```diff
@@ -24,6 +25,16 @@
   
   private AccountEntity accountEntity;
   
+  private CustomerAccountDTO negativeCustomerAccountDTO =
```
Comment (ssuchkov) [inline src/test/java/com/somn/service/AccountEntityServiceImplTest.java]: It's better use names like 'correct' 'incorrect' here

File: src/main/java/com/somn/dto/CustomerAccountDTO.java [hunk 2]
```diff
@@ -7,15 +7,9 @@
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
-public class AccountDTO {
+public class CustomerAccountDTO {
```
Comment (ssuchkov) [inline src/main/java/com/somn/dto/CustomerAccountDTO.java]: Is it possible to extract super class here ?

File: src/main/java/com/somn/mappers/AccountantAccountMapper.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.List;
+
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper
+public interface AccountantAccountMapper {
+  @Mapping(source = "id", target = "id")
+  @Mapping(source = "accountStatus", target = "accountStatus")
```
Comment (ssuchkov) [inline src/main/java/com/somn/mappers/AccountantAccountMapper.java]: do we need this explicit mapping?

File: src/main/java/com/somn/service/CustomerEntityServiceImpl.java [hunk 4]
```diff
@@ -1,25 +1,41 @@
...
+import java.util.Set;
 
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.security.core.userdetails.UserDetails;
 import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class CustomerEntityServiceImpl implements CustomerEntityService {
```
Comment (ssuchkov) [inline src/main/java/com/somn/service/CustomerEntityServiceImpl.java]: We can avoid series of auto wires by declaring final fields and using Lombok @RequiredArgsConstructor

File: src/main/java/com/somn/dto/UserDTO.java [hunk 5]
```diff
@@ -13,8 +12,7 @@
 public class UserDTO {
   private Long id;
   private String firstName;
-  private String encryptedPassword;
+  private String password;
```
Comment (ssuchkov) [inline src/main/java/com/somn/dto/UserDTO.java]: why it became password?
I believe it still encrypted

Comment (evakule) [reply src/main/java/com/somn/dto/UserDTO.java]: Ok

File: src/main/java/com/somn/service/CustomerEntityServiceImpl.java (hunk 4 above)
Comment (evakule) [reply src/main/java/com/somn/service/CustomerEntityServiceImpl.java]: Ok

File: src/main/java/com/somn/mappers/AccountantAccountMapper.java (hunk 3 above)
Comment (evakule) [reply src/main/java/com/somn/mappers/AccountantAccountMapper.java]: Let`s see

File: src/test/java/com/somn/service/AccountEntityServiceImplTest.java (hunk 1 above)
Comment (evakule) [reply src/test/java/com/somn/service/AccountEntityServiceImplTest.java]: Ok

File: src/main/java/com/somn/dto/CustomerAccountDTO.java (hunk 2 above)
Comment (evakule) [reply src/main/java/com/somn/dto/CustomerAccountDTO.java]: Ok

**Pull request merged**